src/components/Reports/Reports.js

```diff
@@ -15,54 +15,47 @@ import { Helmet } from 'react-helmet';
 import { Loading } from '../Loading/Loading';
 import { addDays } from '../../utils';
 
-class Reports extends React.Component {
-  /**
-   * @param { Object } props
-   * @param { import('../../services/pure-di').AppServices } props.dependencies
-   */
-  constructor({ dependencies: { datahubClient, appConfiguration } }) {
-    super();
+const periodSelectedDaysDefault = 7;
 
-    this.datahubClient = datahubClient;
-    this.appConfiguration = appConfiguration;
+const Reports = ({ dependencies: { datahubClient } }) => {
+  const [state, setState] = useState({ periodSelectedDays: periodSelectedDaysDefault });
 
-    this.state = {
-      domains: null,
-      domainSelected: null,
-      periodSelectedDays: 7,
-      dateTo: null,
-      dateFrom: null,
-    };
-
-    this.changeDomain = this.changeDomain.bind(this);
-    this.changePeriod = this.changePeriod.bind(this);
-  }
+  const changeDomain = async (name) => {
+    const domainFound = state.domains.find((item) => item.name === name);
+    setState((prevState) => ({ ...prevState, domainSelected: domainFound }));
+  };
 
-  async componentDidMount() {
-    const domains = await this.datahubClient.getAccountDomains();
-    const domainSelected = domains.length ? domains[0] : null;
+  const changePeriod = (days) => {
     const now = new Date();
-    const dateFrom = addDays(now, parseInt(this.state.periodSelectedDays) * -1);
-    this.setState({
-      domains: domains,
-      domainSelected: domainSelected,
+    const dateFrom = addDays(now, days * -1);
+    setState((prevState) => ({
+      ...prevState,
+      periodSelectedDays: days,
       dateFrom: dateFrom,
       dateTo: now,
-    });
-  }
-
-  changeDomain = async (name) => {
-    const domainFound = this.state.domains.find((item) => item.name === name);
-    this.setState({ domainSelected: domainFound });
+    }));
   };
 
-  changePeriod = (days) => {
-    const now = new Date();
-    const dateFrom = addDays(now, days * -1);
-    this.setState({ periodSelectedDays: days, dateFrom: dateFrom, dateTo: now });
-  };
+  useEffect(() => {
+    const fetchData = async () => {
+      const domains = await datahubClient.getAccountDomains();
+      if (domains) {
+        const now = new Date();
+        const dateFrom = addDays(now, periodSelectedDaysDefault * -1);
```

Comment:
What do you think about adding _periodSelectedDays_ as a dependency and update it declaratively?